src/Eval.hs

```diff
@@ -163,11 +163,13 @@ eval env xobj =
                Left err -> return (Left err)
 
         [defnExpr@(XObj Defn _ _), name, args@(XObj (Arr a) _ _), body] ->
-            if all isSym a
-              then specialCommandDefine xobj
-              else return (makeEvalError ctx Nothing ("`defn` requires all arguments to be symbols, but it got `" ++ pretty args ++ "`") (info xobj))
-            where isSym (XObj (Sym _ _) _ _) = True
-                  isSym _ = False
+            case (obj name) of
+              (Sym _ _) -> if all isSym a
+                           then specialCommandDefine xobj
+                           else return (makeEvalError ctx Nothing ("`defn` requires all arguments to be symbols, but it got `" ++ pretty args ++ "`") (info xobj))
+                        where isSym (XObj (Sym _ _) _ _) = True
```

Comment:
We could constrain this even more, since the `SymPath` module part must be empty (you couldn’t use `Module.x`, for instance), but that would probably require a different error message and make this a tad more complex.

Reply:
OK, I'll merge this and let you add that separately later!